src/Yarhl/IO/DataStream.cs

```diff
@@ -639,6 +639,41 @@ public bool Compare(DataStream otherStream)
             return result;
         }
 
+        /// <summary>
+        /// Creates a substream starting in a defined position.
+        /// </summary>
+        /// <returns>The substream defined by offset and length parameters.</returns>
+        /// <param name="start">Defined starting position.</param>
+        public DataStream Slice(long start)
+        {
+            if (start < 0 || start > Length)
+                throw new ArgumentOutOfRangeException(nameof(start));
+
+            DataStream slicedStream = new DataStream();
```

Comment:
We should call the constructor to create a substream instead of creating a new copy into memory. This would prevent allocating memory and it will be much faster.